src/aspire/abinitio/commonline_d2.py

```diff
@@ -0,0 +1,1888 @@
+    def _saff_kuijlaars(N):
+        """
+        Generates N vertices on the unit sphere that are approximately evenly distributed.
+
+        This implements the recommended algorithm in spherical coordinates
+        (theta, phi) according to "Distributing many points on a sphere"
+        by E.B. Saff and A.B.J. Kuijlaars, Mathematical Intelligencer 19.1
+        (1997) 5--11.
+
+        :param N: Number of vertices to generate.
+
+        :return: Nx3 array of vertices in cartesian coordinates.
+        """
+        k = np.arange(1, N + 1)
+        h = -1 + 2 * (k - 1) / (N - 1)
+        theta = np.arccos(h)
+        phi = np.zeros(N)
+
+        for i in range(1, N - 1):
+            phi[i] = (phi[i - 1] + 3.6 / (np.sqrt(N * (1 - h[i] ** 2)))) % (2 * np.pi)
+
+        # Spherical coordinates
+        x = np.sin(theta) * np.cos(phi)
+        y = np.sin(theta) * np.sin(phi)
+        z = np.cos(theta)
+
+        mesh = np.column_stack((x, y, z))
+
+        return mesh
+
+    @staticmethod
+    def _mark_equators(sphere_grid, eq_filter_angle):
+        """
+        :param sphere_grid: Nx3 array of vertices in cartesian coordinates.
+        :param eq_filter_angle: Angular distance from equator to be marked as
+            an equator point.
+
+        :returns:
+            - eq_idx, a boolean mask for equator indices.
+            - eq_class, n_rots length array of values indicating equator class.
+        """
+        # Project each vector onto xy, xz, yz planes and measure angular distance
+        # from each plane.
+        n_rots = len(sphere_grid)
+        angular_dists = np.zeros((n_rots, 3), dtype=sphere_grid.dtype)
+
+        # Distance from z-axis equator.
+        proj_xy = sphere_grid.copy()
+        proj_xy[:, 2] = 0
+        proj_xy /= np.linalg.norm(proj_xy, axis=1)[:, None]
+        angular_dists[:, 0] = np.sum(sphere_grid * proj_xy, axis=-1)
+
+        # Distance from y-axis equator.
+        proj_xz = sphere_grid.copy()
+        proj_xz[:, 1] = 0
+        proj_xz /= np.linalg.norm(proj_xz, axis=1)[:, None]
+        angular_dists[:, 1] = np.sum(sphere_grid * proj_xz, axis=-1)
+
+        # Distance from x-axis equator.
+        proj_yz = sphere_grid.copy()
+        proj_yz[:, 0] = 0
+        proj_yz /= np.linalg.norm(proj_yz, axis=1)[:, None]
+        angular_dists[:, 2] = np.sum(sphere_grid * proj_yz, axis=-1)
+
+        # Mark all views close to an equator.
+        eq_min_dist = np.cos(eq_filter_angle * np.pi / 180)
+        n_eqs = np.sum(angular_dists > eq_min_dist, axis=1)
+        eq_idx = n_eqs > 0
+
+        # Classify equators.
+        # 0 -> non-equator view
+        # 1 -> z equator
+        # 2 -> y equator
+        # 3 -> x equator
+        # 4 -> z top view
+        # 5 -> y top view
+        # 6 -> x top view
+        eq_class = np.zeros(n_rots)
+        top_view_idx = n_eqs > 1
+        top_view_class = np.argmin(angular_dists[top_view_idx] > eq_min_dist)
+        eq_class[top_view_idx] = top_view_class + 4
```

Comment:
Added a comment. This was actually not working properly because I did not designate an axis for the `argmin`. We end up tossing top views later so the bug wasn't causing a noticeable problem. 

This raises the obvious question of why do we even classify them. The answer is because that is what the original author did.